### PR TITLE
Temporal: Test order of property accesses

### PR DESCRIPTION
**Pull request by ptomato**
This contains tests for the normative PR
https://github.com/tc39/proposal-temporal/pull/2437, which is to be
presented for consensus to TC39 in the upcoming plenary meeting. That PR
changes the observable order of property accesses to be alphabetical where
possible.

This PR is based on https://github.com/tc39/test262/pull/3712, so review
that first and rebase this one when it is merged.

### CODE REVIEW

**Comment by cjtenny** [comment]
Lgtm, straightforward :)

**Review by Ms2ger — APPROVED** [review]
Pending approval of the spec change

**Comment by ptomato** [comment]
The corresponding change achieved consensus in the TC39 plenary today.

I rebased this after #3712 got merged and it cancelled the existing codeowner approval from Ms2ger; rubberstamp plz?

**Review by Ms2ger — APPROVED** [review]

**Pull request merged**
